packages/http/src/types.ts

```diff
@@ -500,6 +510,53 @@ export interface HttpOperationPart {
   readonly multi: boolean;
 }
 
+/**
+ * The type of an HTTP body that is a file upload or download.
+ */
+export interface HttpOperationFileBody extends HttpOperationBodyBase {
+  readonly bodyKind: "file";
+  /**
+   * The model type of the body that is or extends `Http.File`.
+   */
+  readonly type: Model;
+
+  /**
+   * Whether the file contents should be represented as a string or raw byte stream.
+   *
+   * True if the `contents` property is a `string`, `false` if it is `bytes`.
+   *
+   * Emitters may choose to represent textual files as strings or streams of textual characters.
+   * If this property is `false`, emitters must expect that the contents may contain non-textual
+   * data.
+   */
+  readonly isText: boolean;
+
+  /**
+   * The list of inner media types of the file. In other words, what kind of files can be returned.
+   *
+   * This is determined by the `contentType` property of the file model.
+   */
+  readonly contentTypes: string[];
+
+  /**
+   * The `contentType` property.
+   */
+  readonly contentTypeProperty: ModelProperty;
+
+  /**
+   * The filename property.
+   */
+  readonly filename: ModelProperty;
+
+  /**
+   * The `contents` property.
+   */
+  readonly contents: ModelProperty & { type: Scalar & { name: "bytes" | "string" } };
```

Comment:
Why we flatten all inner properties to operation level? Can we still get original model name like `SpecFile`?

We are talking about we may accept more properties in future, then what will it look like?

```ts
model SpecFile extends File {
  @header foo: string;
}
```

Reply:
I'm not sure what you mean by "flattening the inner properties" here. If the file is `SpecFile` you will see `SpecFile` in the `type` property, and I will add some tests to ensure this works correctly. The docs for `type` say "The model type that **is or extends** `Http.File`.", so you will have a reference to the type that was detected as a File.